### PR TITLE
Update Components to use LoggerMessage

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by davidfowl** [comment]
Does blazor have any tests that very logs?

**Comment by pranavkm** [comment]
> Does blazor have any tests that very logs?

Probably not? I think there are a few that verify error cases, but we rarely verify the vast majority of log messages.

File: src/Components/Server/src/Circuits/CircuitHost.cs [hunk 1]
```diff
@@ -641,322 +635,146 @@ private async Task TryNotifyClientErrorAsync(IClientProxy client, string error,
...
+            [LoggerMessage(111, LogLevel.Error, "Unhandled exception in circuit '{CircuitId}'.", EventName = "CircuitUnhandledException")]
+            public static partial void CircuitUnhandledException(ILogger logger, CircuitId circuitId, Exception exception);
+
+            [LoggerMessage(114, LogLevel.Debug, "Failed to transmit exception to client in circuit '{CircuitId}'.", EventName = "CircuitTransmitErrorFailed")]
+            public static partial void CircuitTransmitErrorFailed(ILogger logger, CircuitId circuitId, Exception exception);
+
+            [LoggerMessage(204, LogLevel.Debug, "There was an error invoking 'Microsoft.JSInterop.DotNetDispatcher.EndInvoke'.", EventName = "EndInvokeDispatchException")]
+            public static partial void EndInvokeDispatchException(ILogger logger, Exception ex);
+
+            [LoggerMessage(205, LogLevel.Debug, "The JS interop call with callback id '{AsyncCall}' with arguments {Arguments}.", EventName = "EndInvokeJSFailed")]
```
Comment (pranavkm) [inline src/Components/Server/src/Circuits/CircuitHost.cs:695]: This log statement changed. It used to previously say:

> "The JS interop call with callback id '{AsyncCall}' failed with error '{Error}'."

but Error is not value that appears in the code. The calling code does say it should be safe to include the Arguments: in the log message, so this was probably a wrong log statement all along 

https://github.com/dotnet/aspnetcore/blob/832ea98014f9a49fceecf6b34e8c8b4c16b48ff5/src/Components/Server/src/Circuits/CircuitHost.cs#L371

File: src/Components/Components/src/RenderTree/Renderer.Log.cs [hunk 2]
```diff
@@ -1,69 +1,68 @@
...
                 }
             }
 
+            [LoggerMessage(3, LogLevel.Debug, "Rendering component {ComponentId} of type {ComponentType}", EventName = "RenderingComponent", SkipEnabledCheck = true)]
+            private static partial void RenderingComponent(ILogger logger, int componentId, Type componentType);
+
             public static void RenderingComponent(ILogger logger, ComponentState componentState)
             {
+
                 if (logger.IsEnabled(LogLevel.Debug)) // This is almost always false, so skip the evaluations
```
Comment (TanayParikh) [inline src/Components/Components/src/RenderTree/Renderer.Log.cs:40]:
```suggestion
                if (logger.IsEnabled(LogLevel.Debug)) // This is almost always false, so skip the evaluations
```

File: src/Components/Server/src/Circuits/CircuitHost.cs [hunk 3]
```diff
@@ -641,322 +635,146 @@ private async Task TryNotifyClientErrorAsync(IClientProxy client, string error,
...
+            public static partial void SendDotNetStreamException(ILogger logger, long id, Exception ex);
+
+            [LoggerMessage(200, LogLevel.Debug, "Failed to parse the event data when trying to dispatch an event.", EventName = "DispatchEventFailedToParseEventData")]
+            public static partial void DispatchEventFailedToParseEventData(ILogger logger, Exception ex);
+
+            [LoggerMessage(201, LogLevel.Debug, "There was an error dispatching the event '{EventHandlerId}' to the application.", EventName = "DispatchEventFailedToDispatchEvent")]
+            public static partial void DispatchEventFailedToDispatchEvent(ILogger logger, string eventHandlerId, Exception ex);
+
+            [LoggerMessage(202, LogLevel.Debug, "Invoking instance method '{MethodIdentifier}' on instance '{DotNetObjectId}' with callback id '{CallId}'.", EventName = "BeginInvokeDotNet")]
+            private static partial void BeginInvokeDotNet(ILogger logger, string methodIdentifier, long dotNetObjectId, string callId);
```
Comment (TanayParikh) [inline src/Components/Server/src/Circuits/CircuitHost.cs:720]: Nit; can we move these up so the ids are in order?

File: src/Components/Server/src/Circuits/CircuitHost.cs [hunk 4]
```diff
@@ -641,322 +635,146 @@ private async Task TryNotifyClientErrorAsync(IClientProxy client, string error,
...
                 }
                 else
                 {
-                    _beginInvokeDotNetInstance(logger, methodIdentifier, dotNetObjectId, callId, null);
+                    BeginInvokeDotNet(logger, methodIdentifier, dotNetObjectId, callId);
                 }
             }
 
+            [LoggerMessage(203, LogLevel.Debug, "Failed to invoke instance method '{MethodIdentifier}' on instance '{DotNetObjectId}' with callback id '{CallId}'.", EventName = "BeginInvokeDotNetFailed")]
+            private static partial void BeginInvokeDotNetFailed(ILogger logger, string methodIdentifier, long dotNetObjectId, string callId, Exception exception);
```
Comment (TanayParikh) [inline src/Components/Server/src/Circuits/CircuitHost.cs:738]: ditto

File: src/Components/Server/src/Circuits/CircuitRegistry.cs [hunk 5]
```diff
@@ -362,164 +359,58 @@ public DisconnectedCircuitEntry(CircuitHost circuitHost, CancellationTokenSource
...
 
-            public static void FailedToReconnectToCircuit(ILogger logger, CircuitId circuitId, Exception exception = null) =>
-                _failedToReconnectToCircuit(logger, circuitId, exception);
+            [LoggerMessage(107, LogLevel.Debug, "Failed to reconnect to a circuit with id {CircuitId}.", EventName = "FailedToReconnectToCircuit")]
+            public static partial void FailedToReconnectToCircuit(ILogger logger, CircuitId circuitId, Exception exception = null);
 
-            public static void ReconnectionSucceeded(ILogger logger, CircuitId circuitId) =>
-                _reconnectionSucceeded(logger, circuitId, null);
+            [LoggerMessage(115, LogLevel.Debug, "Reconnect to circuit with id {CircuitId} succeeded.", EventName = "FailedToReconnectToCircuit")]
+            public static partial void ReconnectionSucceeded(ILogger logger, CircuitId circuitId);
```
Comment (TanayParikh) [inline src/Components/Server/src/Circuits/CircuitRegistry.cs:392]: Nit; move to end due to id.

File: src/Components/Server/src/Circuits/RemoteRenderer.cs [hunk 6]
```diff
@@ -341,131 +339,43 @@ private async Task CaptureAsyncExceptions(Task task)
...
-                    LogLevel.Debug,
-                    EventIds.ReceivedDuplicateBatchAcknowledgement,
-                    "Received a duplicate ACK for batch id '{IncomingBatchId}'.");
-
-                _fullUnacknowledgedRenderBatchesQueue = LoggerMessage.Define(
-                    LogLevel.Debug,
-                    EventIds.FullUnacknowledgedRenderBatchesQueue,
-                    "The queue of unacknowledged render batches is full.");
-            }
+            [LoggerMessage(103, LogLevel.Information, "Sending data for batch failed: {Message}", EventName = "SendBatchDataFailed")]
```
Comment (TanayParikh) [inline src/Components/Server/src/Circuits/RemoteRenderer.cs:344]: Ordering

**Review by TanayParikh — APPROVED** [review]
LGTM; just a couple of nits with regard to log ordering based on ids.

**Pull request merged**